ipn/config: add RelayServerPort and RelayServerStaticEndpoints to config file

rajsinghtech opened the pull request:
Adds relay server config fields to ConfigVAlpha so they can be set via config file.

Fixes #18299

jwhited commented:
cc @tailscale/k8s-devs, who are the SMEs of this config file machinery

davidsbond commented:
Reopening this as we have a design for peer relays in kube now and will want these config values.

jwhited reviewed (changes requested):
Let's add a test demonstrating this works as expected. Make sure to include disablement  cases.

rajsinghtech committed:
ipn/config: add RelayServerPort and RelayServerStaticEndpoints to config file

Adds relay server config fields to ConfigVAlpha so they can be set
via the config file. Includes tests covering the ToPrefs conversion
(set, nil, zero-port, and disablement cases) and config-file reload
(enable, disable, change).

Fixes #18299

Signed-off-by: Raj Singh <raj@tailscale.com>

davidsbond commented on ipn/ipnlocal/local_test.go:
```diff
@@ -7490,6 +7491,78 @@ func TestConfigFileReload(t *testing.T) {
+			updated: &conffile.Config{
+				Parsed: ipn.ConfigVAlpha{
+					Version:                    "alpha0",
+					RelayServerPort:            ptr.To(uint16(12345)),
```
IIRC you should be able to use Go's builtin `new` function for this now.

rajsinghtech replied:
done!

davidsbond commented on ipn/conf_test.go:
```diff
@@ -64,3 +68,122 @@ func TestConfigVAlpha_ToPrefs_AdvertiseRoutes(t *testing.T) {
+		for _, s := range strs {
+			n, err := netip.ParseAddrPort(s)
+			if err != nil {
+				panic(err)
```
We have access to `t` here, so instead of panicking could we use `Fatalf`?

rajsinghtech replied:
done!

davidsbond commented:
@jwhited do you have anything else for this? We'll be wanting to use this for peer relays in Kube.

bradfitz commented:
You had a commit message + issue number in here originally, but you merged this without one:

```
commit b64209b24800ca1254fb884b848fad256cd99460 (HEAD -> main, origin/main, origin/HEAD)
Author: Raj Singh <raj@tailscale.com>
Date:   Fri Jun 26 15:45:29 2026 -0500

    ipn/config: add RelayServerPort and RelayServerStaticEndpoints to config file (#18300)
```

How'd that happen?

rajsinghtech commented:
> You had a commit message + issue number in here originally, but you merged this without one:
> 
> ```
> commit b64209b24800ca1254fb884b848fad256cd99460 (HEAD -> main, origin/main, origin/HEAD)
> Author: Raj Singh <raj@tailscale.com>
> Date:   Fri Jun 26 15:45:29 2026 -0500
> 
>     ipn/config: add RelayServerPort and RelayServerStaticEndpoints to config file (#18300)
> ```
> 
> How'd that happen?

im as confused as you are, I believe I clicked squash and merge from my phone fwiw. where is my body?
